docs/developer/extending/apps/developing-apps/app-error-codes.mdx

```diff
@@ -0,0 +1,225 @@
+  - If your app returned an HTTP error (like 404 or 500), check its logs to understand the cause of the status code.
+  - **Saleor Cloud users**: If you suspect a network issue on Saleor's side after verifying your URL is public, contact Saleor Support.
+  - **Self-hosted users**: Check your Saleor instance's network configuration and timeout settings.
+    1. From your Saleor instance's environment, perform network diagnostics (e.g., `curl` or `ping` the `manifestUrl`) to check reachability and the HTTP response.
+    2. Review firewall configurations for outgoing connections from your Saleor instance, if applicable.
+    3. Inspect background worker logs for network request exception details or a generic connection failure message.
+
+
+
+## `AppErrorCode.INVALID_MANIFEST_FORMAT`
+This error means the manifest has invalid structure. This can include issues like the content not being valid JSON or other structural problems. It often summarizes multiple specific validation errors found within the manifest content.
+
+- 🧐 **Cause**: The manifest file does not adhere to the expected structure, contains formatting issues, or is not valid JSON.
+- 💡 **Debugging Tips**:
+  - Before installation, validate your app manifest against Saleor's schema. Refer to the [App Manifest documentation](/developer/extending/apps/architecture/manifest.mdx) for schema details. Using a JSON linter can also catch basic syntax errors.
+  - If you encounter a JSON decode error:
+    1. Obtain the raw content fetched from the `manifestUrl` (e.g., using `curl`).
+    2. Validate this content with a JSON linter.
+  - Review the full manifest content carefully for any structural issues.
+  - Examine the full error returned by Saleor. It often points to specific problematic fields in the manifest and explains why they are problematic.
+  - Consider using Saleor's [`app-sdk`](/developer/extending/apps/developing-apps/app-sdk/overview.mdx) to construct the manifest, as it helps adhere to the documented structure.
+
+## `AppErrorCode.REQUIRED`
+This error means the manifest is missing one or more mandatory top-level fields.
+
+- 🧐 **Cause**: Mandatory top-level fields (e.g., `id`, `version`, `name`, `tokenTargetUrl`) are missing from the manifest.
+- 💡 **Debugging Tips**:
+  - Inspect the full manifest content.
+  - Check error details, which should indicate the specific required field(s) that are missing.
+  - Refer to Saleor's [App Manifest documentation](/developer/extending/apps/architecture/manifest.mdx) to ensure all required fields are present.
+
+## `AppErrorCode.UNIQUE`
+This error means the `id` field (identifier) in the manifest is already used by another app in your Saleor instance.
+
+- 🧐 **Cause**: The app identifier specified in the manifest is not unique within the Saleor instance.
+- 💡 **Debugging Tips**:
+  - Check if an app with the same `id` is already installed.
```

Comment:
I think that we also take into account the ongoing installation - maybe worth to mention it here

Reply:
yup you were right, updated that part